app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,28 @@
+class SessionsController < ApplicationController
+  def new
+    # shows a view with OAuth sign-in link
+  end
+
+  def create
+    # accepts OAuth information from Spotify, finds or creates a User account, and sets user_id in session
```

Comment:
Unless it's not clear from the code what the function is doing, comments aren't necessary in the final PR